### DISCUSSION TITLE
Run format_on_save command in subdirectory

### DISCUSSION THREAD

**Discussion by atomkirk** [zed-industries/zed#7672] [Feature Requests]
I've got a project folder that has a bunch of subdirectorys like `api`, `infra`, `ios` etc. A monorepo really. I like to open the whole thing in zed so I have access to everything. I don't know how to configure my elixir formatter so that `mix` runs inside `api`. Right now it complains it can't find a mix project because im guessing it tries to run it from the root.

```
{
  "language_overrides": {
    "Elixir": {
      "format_on_save": {
        "external": {
          "command": "mix",
          "arguments": [
            "format",
            "{buffer_path}"
          ]
        }
      }
    }
  }
}
```

Any ideas for me?

I absolutely love this. If I can get this working, this is my new editor.

**Comment by atomkirk** [comment]
Same thing with the language server. Doesn't work if my elixir project is in a subdirectory.

**Reply by cschmatzler** [reply]
Did you find a solution?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
